Training/Setup/Patch/Data/AddCustomerAttributeCreditHold.php

```diff
@@ -0,0 +1,115 @@
+<?php
+/**
+ * Codifi_Training
+ *
+ * @copyright   Copyright (c) 2021 Codifi
+ * @author      Pavel Zelenevich <pzelenevich@codifi.me>
+ */
+
+namespace Codifi\Training\Setup\Patch\Data;
+
+use Magento\Framework\Setup\Patch\DataPatchInterface;
+use Magento\Framework\Setup\Patch\PatchRevertableInterface;
+use Magento\Framework\Setup\ModuleDataSetupInterface;
+use Magento\Customer\Setup\CustomerSetupFactory;
+use Magento\Customer\Setup\CustomerSetup;
+use Magento\Eav\Model\Entity\Attribute\SetFactory as AttributeSetFactory;
+
+class AddCustomerAttributeCreditHold implements DataPatchInterface, PatchRevertableInterface
+{
+    /**
+     * @var ModuleDataSetupInterface
+     */
+    private $moduleDataSetup;
+    /**
+     * @var CustomerSetup
+     */
+    private $customerSetupFactory;
+    /**
+     * @var AttributeSetFactory
+     */
+    protected $attributeSetFactory;
+
+    /**
+     * Constructor
+     *
+     * @param ModuleDataSetupInterface $moduleDataSetup
+     * @param CustomerSetupFactory $customerSetupFactory
+     */
+    public function __construct(
+        ModuleDataSetupInterface $moduleDataSetup,
+        CustomerSetupFactory $customerSetupFactory,
+        AttributeSetFactory $attributeSetFactory
+    ) {
+        $this->moduleDataSetup = $moduleDataSetup;
+        $this->customerSetupFactory = $customerSetupFactory;
+        $this->attributeSetFactory = $attributeSetFactory;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function apply()
+    {
+        $this->moduleDataSetup->getConnection()->startSetup();
+        /** @var CustomerSetup $customerSetup */
+        $customerSetup = $this->customerSetupFactory->create(['setup' => $this->moduleDataSetup]);
+        $customerSetup->addAttribute(
+            \Magento\Customer\Model\Customer::ENTITY,
+            'credit_hold',
+            [
+                'type' => 'int',
+                'label' => 'Credit Hold',
+                'input' => 'select',
+                'source' => \Magento\Eav\Model\Entity\Attribute\Source\Table::class,
```

Comment:
Review implementation of 'source' in vendor/magento/module-rma/Setup/Patch/Data/AddRmaAttributes.php:160
Magento\Rma\Model\Product\Source